EasyReflectometry/parameter_utils.py

```diff
@@ -0,0 +1,18 @@
+from copy import deepcopy
+from numbers import Number
+from typing import Union
+
+from easyCore.Objects.ObjectClasses import Parameter
+
+
+def get_as_parameter(value: Union[Parameter, Number, None], name, default_dict: dict[str, str]) -> Parameter:
+    # Should leave the passed dictionary unchanged
+    default_dict = deepcopy(default_dict)
```

Comment:
Added comments in the code comments.

The dictionary only serves to provide the needed data to create a parameter and it should be completely "decoupled" from the created parameter.  In the old setup we did not do an explicit deepcopy for `default` since the dictionary was left unchanged (on `del` statement)